File: docs/api/spring/com/shoptourr/api/v1/dto/export/ExportDtos.java
package com.shoptourr.api.v1.dto.export;

import jakarta.validation.constraints.NotNull;

import java.time.Instant;
import java.util.UUID;

/** Export PDF / CSV — always async job. */
public final class ExportDtos {

    private ExportDtos() {}

    public enum ExportFormat {
        PDF, CSV
    }

    public enum ExportJobStatus {
        QUEUED, RUNNING, READY, FAILED, EXPIRED
    }

    public record CreateExportRequest(
            @NotNull ExportFormat format,
            /** Include Tax Free worksheet section (PDF). */
            boolean includeTaxFree,
            boolean includeDiary
    ) {}

    public record ExportJobDto(
            UUID id,
            UUID tripId,
            ExportFormat format,
            ExportJobStatus status,
            String downloadUrl,
            Instant expiresAt,
            String errorCode,
            Instant createdAt,
            Instant finishedAt
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/data/remote/dto/export/ExportDtos.kt
package com.example.shoptourr.data.remote.dto.export

enum class ExportFormat { PDF, CSV }

enum class ExportJobStatus { QUEUED, RUNNING, READY, FAILED, EXPIRED }

data class CreateExportRequest(
    val format: ExportFormat,
    val includeTaxFree: Boolean = true,
    val includeDiary: Boolean = false,
)

data class ExportJobDto(
    val id: String,
    val tripId: String,
    val format: ExportFormat,
    val status: ExportJobStatus,
    val downloadUrl: String? = null,
    val expiresAt: String? = null,
    val errorCode: String? = null,
    val createdAt: String,
    val finishedAt: String? = null,
)
